docs/gettingstarted.md

```diff
@@ -1,9 +1,13 @@
-Now that you have `gdbgui` installed, all you need to do is run
+Before running `gdbgui`, you should compile your program with debug symbols and a lower level of optimization, so code isn't optimized out before runtime. To include debug symbols with `gcc` or `rustc`, this means using `-g`. To disable most optimizations in gdcc `gcc` use the `-O0` flag.
```

Comment:
I recommend changing this: 
```
To include debug symbols with `gcc` or `rustc`, this means using `-g`. To disable most optimizations in gdcc `gcc` use the `-O0` flag.
```
To:
```
To include debug symbols with `gcc` or `rustc`, use `-g` to enable debugging symbols, or even better, `-ggdb` to optimize the debugging symbols for the `gdb` debugger. To disable most optimizations in `gcc` use the `-O0` flag. The `-O0` optimization level is preferred over `-Og` because even `-Og` can sometimes optimize out variables you'd like to inspect, [as shown here](https://stackoverflow.com/a/63386263/4561887). In summary, we recommend you add both `-ggdb` and `-O0` to build the proper debugging symbols.
```

Reply:
Thanks, I updated it. I changed to `-ggdb` for gcc and added more instructions for rust. I am going hold off on adding more details about `-Og` since I am only trying to document how to get started.